modules/RTCStats/RTCStats.ts

```diff
@@ -0,0 +1,189 @@
+import rtcstatsInit from '@jitsi/rtcstats/rtcstats';
+import traceInit from '@jitsi/rtcstats/trace-ws';
+import EventEmitter from 'events';
+
+import { 
+    CONFERENCE_JOINED, 
+    CONFERENCE_LEFT, 
+    CONFERENCE_UNIQUE_ID_SET 
+} from '../../JitsiConferenceEvents';
+import JitsiConference from '../../JitsiConference';
+
+const RTC_STATS_WC_DISCONNECTED = 'rtcstats_ws_disconnected';
+const RTC_STATS_PC_EVENT = 'rtstats_pc_event';
+
+/**
+ * Filter out RTCPeerConnection that are created by callstats.io.
+ *
+ * @param {*} config - Config object sent to the PC c'tor.
+ * @returns {boolean}
+ */
+function connectionFilter(config) {
+    if (config?.iceServers[0] && config.iceServers[0].urls) {
+        for (const iceUrl of config.iceServers[0].urls) {
+            if (iceUrl.includes('callstats.io')) {
+                return true;
+            }
+        }
+    }
+}
+
+/**
+ * RTCStats Singleton that is initialized only once for the lifetime of the app, subsequent calls to init will be ignored.
+ * Config and conference changes are handled by the start method.
+ */
+class RTCStats {
+    private _initialized: boolean = false;
+    private trace: any = null;
+    public events: EventEmitter = new EventEmitter();
+
+    /**
+     * RTCStats "proxies" WebRTC functions such as GUM and RTCPeerConnection by rewriting the global objects.
+     * The proxies will then send data to the rtcstats server via the trace object.
+     * The initialization procedure must be called once when lib-jitsi-meet is loaded.
+     * 
+     * @param initConfig initial config for rtcstats.
+     * @returns {void}
+     */
+    init(initConfig: any) {        
+        const {
+            analytics: {
+                rtcstatsUseLegacy: useLegacy = false,
+                rtcstatsPollInterval: pollInterval= 10000,
+                rtcstatsSendSdp: sendSdp = false,
+                rtcstatsEnabled = false
+            } = {}
+        } = initConfig;
+
+        // If rtcstats is not enabled or already initialized, do nothing.
+        // Calling rtcsatsInit multiple times will cause the global objects to be rewritten multiple times,
+        // with unforseseen consequences.
+        if (!rtcstatsEnabled || this._initialized) return;
+        
+        rtcstatsInit( 
```

Comment:
Can this fail?

Reply:
It could, would you rather I swallow and log the exception or let it fail?

Reply:
I guess it's fine since the outcome is the same.